brian2/codegen/runtime/weave_rt/templates/synapses_create_array.cpp

```diff
@@ -19,20 +19,20 @@ _arg_tuple_new[0] = _new_num_synapses;
 int *_synaptic_pre_data = (int*)(((PyArrayObject*)(PyObject*){{_dynamic__synaptic_pre}}.attr("data"))->data);
 int *_synaptic_post_data = (int*)(((PyArrayObject*)(PyObject*){{_dynamic__synaptic_post}}.attr("data"))->data);
 
-for (int _idx=0; _idx<_numsources; _idx++) {
+for (size_t _idx=0; _idx<_numsources; _idx++) {
     {# After this code has been executed, the arrays _real_sources and
        _real_variables contain the final indices. Having any code here it all is
        only necessary for supporting subgroups #}
     {{vector_code|autoindent}}
-    _synaptic_pre_data[_idx + _old_num_synapses] = _real_sources;
-    _synaptic_post_data[_idx + _old_num_synapses] = _real_targets;
+    _synaptic_pre_data[_idx + {{N}}] = _real_sources;
+    _synaptic_post_data[_idx + {{N}}] = _real_targets;
```

Comment:
What's the reason for this change? (Used again later.)

Reply:
It's just that previously we set `_old_num_synapses = {{N}}` but never change either of them, so I thought we could just use `{{N}}` directly.